fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/properties/RecordLayerPropertyKey.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * RecordLayerPropertyKey.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.properties;
+
+import com.apple.foundationdb.annotation.API;
+
+import javax.annotation.Nonnull;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+/**
+ * A key for a property that adopter configures for Record Layer.
+ * This is not supposed to be defined by adopter outside Record Layer.
+ * The name, value type, and its default value are defined in this key, which are immutable.
+ * Call {@link #buildValue(Supplier)} with a {@link Supplier} of the value to build a {@link RecordLayerPropertyValue},
+ * to overrides the value for the property, and then put it into {@link RecordLayerPropertyStorage} with using its builder.
```

Comment:
```suggestion
 * to override the value for the property, and then build a {@link RecordLayerPropertyStorage} with the new property value.
```

It's possible this paragraph could use a little more word-smithing, but that would be my suggestion for now

Reply:
Fixed.